runtime/vm/resolvefield.cpp

```diff
@@ -675,7 +675,7 @@ fieldOffsetsStartDo(J9JavaVM *vm, J9ROMClass *romClass, J9Class *superClazz, J9R
 		 */
 		if ((LOCKWORD_NEEDED == lockwordNeeded)||(NO_LOCKWORD_NEEDED == lockwordNeeded)) {
 			if ((NULL != superClazz) && ((UDATA)-1 != superClazz->lockOffset) && (0 == J9CLASS_DEPTH(superClazz))) {
-				U_32 newSuperSize = fieldInfo.getSuperclassFieldsSize() - referenceSize;
+				U_32 newSuperSize = fieldInfo.getSuperclassFieldsSize() - (U_32)referenceSize;
```

Comment:
Would it be better to change the definition of `referenceSize` to be a `U_32`?

Reply:
It could probably be done, but this was the minimal change. It's used in a bunch of other places.

Reply:
If you request the change below, I'll try this one as well. It should be safe when mixed with UDATA math.